pkg/client/client.go

```diff
@@ -406,7 +407,7 @@ func (client *DataSetClient) sendBufferWithRetryPolicy(buf *buffer.Buffer) {
 		} else {
 			err = fmt.Errorf("non recoverable error %w", err)
 			client.onBufferDrop(buf, lastHttpStatus, err)
-			break // exit loop (failed to send buffer)
+			return false // exit loop (failed to send buffer)
```

Comment:
Do we ever return `true` on success? I only see us returning false, but I don't see `true` anywhere, but I could have missed it.

Reply:
We return true here - https://github.com/scalyr/dataset-go/pull/52/files#diff-2895f30116602d8fe6b0545c186d4f81247fa4eaab049dab174c6c91c0036e08R379